tests/integration/bond_test.py

```diff
@@ -132,3 +134,38 @@ def test_add_bond_with_slaves_and_ipv4(eth1_up, eth2_up, setup_remove_bond99):
     netapplier.apply(desired_bond_state)
 
     assertlib.assert_state(desired_bond_state)
+
+
+def test_rollback_for_bond(eth1_up, eth2_up):
+    current_state = netinfo.show()
+    desired_state = {
+                INTERFACES: [
+                    {
+                        'name': 'bond99',
+                        'type': 'bond',
+                        'state': 'up',
+                        'ipv4': {
+                            'enabled': True,
+                            'address': [
+                                {'ip': '192.168.122.250', 'prefix-length': 24}
+                            ]
+                        },
+                        'link-aggregation': {
+                            'mode': 'balance-rr',
+                            'slaves': ['eth1', 'eth2'],
+                            'options':
+                                {'miimon': '140'}
+                        },
+                    }
+                ]
+            }
+
+    desired_state[INTERFACES][0]['invalid_key'] = 'foo'
+
+    with pytest.raises(netapplier.DesiredStateIsNotCurrentError):
+        netapplier.apply(desired_state)
+
+    time.sleep(2)
```

Comment:
Why is the sleep needed here?

Reply:
Sleep needed for let to rollback fully completed, before assert .

Reply:
@EdDev this sounds to me like there is a bug in the rollback code and it should properly wait until the rollback is completed.